### PR TITLE
Fix "Blazor enhanced form handling doesn't honor button `formmethod` attribute"

### LINKED ISSUE #51940: Blazor enhanced form handling doesn't honor button `formmethod` attribute
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Blazor enhanced form handling doesn't honor button `formmethod` attribute

### Expected Behavior

Behavior of enhanced and non-enhanced form should be the same - button's formmethod attribute should override form's method attribute

### Steps To Reproduce

1. Create new Blazor Web project
2. Add this to Home.razor
```html
<form data-enhance id="form1" method="get">
    <input type="submit" formmethod="post" />
</form>
```
3. Click on Submit button and see in browser's Dev tools "Network" tab that method was "GET" not "POST.

### Exceptions (if any)

_No response_

### .NET Version

8

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by surayya-MS**
# Fix "Blazor enhanced form handling doesn't honor button `formmethod` attribute"

Fixes #51940

### CODE REVIEW

File: src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/FormSubmitButtonWithFormmethod.razor [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿@page "/forms/form-submit-button-with-formmethod"
+@using Microsoft.AspNetCore.Components.Forms
+
+<h1>Form Submit Button with formaction</h1>
+
+<form data-enhance method="get" @onsubmit="() => _submitted = true" @formname="someform">
+    <AntiforgeryToken />
```
Comment (halter73) [inline src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/FormSubmitButtonWithFormmethod.razor:7]: Is the AntiforgeryToken still necessary for a get request?

**Review by halter73 — APPROVED** [review]
Is there any way we can assert inside of the `@onsubmit` callback that the request that triggered it was indeed a POST like we expected?

**Comment by surayya-MS** [comment]
> Is there any way we can assert inside of the `@onsubmit` callback that the request that triggered it was indeed a POST like we expected?

I couldn't find a way to do that. 
For this test I checked all three options: 'get', 'post', 'dialog'. The test passes only with 'post', 'get' results in added query to the url, 'dialog' throws an error in browser console "'dialog' is unsupported in no-cors mode"

**Comment by javiercn** [comment]
If method on the form="dialog" or formmethod="dialog" we should not enhance the form.

**Comment by javiercn** [comment]
@SteveSandersonMS fyi.

**Pull request merged**